core/src/main/java/hudson/model/Run.java

```diff
@@ -1801,6 +1801,7 @@ protected final void execute(@Nonnull RunExecution job) {
 		LOGGER.log(Level.SEVERE, "Failed to rotate log",e);
 	    }
         } finally {
+            IOUtils.closeQuietly(listener);
```

Comment:
Logging would be useful here. IIRC there is a method in hudson.Util or other util class